From e94e529aefb28b65165e978daa2bfd89867ee8d0 Mon Sep 17 00:00:00 2001
From: Dmitry Baryshkov <dmitry.baryshkov@oss.qualcomm.com>
Date: Fri, 21 Mar 2025 17:17:52 +0200
Subject: [PATCH] SPIRV-LLVM-Translator: fix handling of built-in SPIRV targer

If the SPIRV-LLVM-Translator is being built together with the LLVM, set
the defines declaring whether SPIRV target is also included into the
LLVM library or not. Otherwise llvm-spirv execution fails because
spirv-ext option is registered twice.

Upstream-Status: Pending [will be submitted upstream after internal clearance]
Signed-off-by: Dmitry Baryshkov <dmitry.baryshkov@oss.qualcomm.com>

Index: llvm-project-20.1.0.src/llvm/projects/SPIRV-LLVM-Translator/CMakeLists.txt
===================================================================
--- llvm-project-20.1.0.src.orig/llvm/projects/SPIRV-LLVM-Translator/CMakeLists.txt
+++ llvm-project-20.1.0.src/llvm/projects/SPIRV-LLVM-Translator/CMakeLists.txt
@@ -116,6 +116,12 @@ if(LLVM_SPIRV_BUILD_EXTERNAL)
     set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE ccache)
     set_property(GLOBAL PROPERTY RULE_LAUNCH_LINK ccache)
   endif()
+else()
+  is_llvm_target_library("SPIRV" spirv_present_result INCLUDED_TARGETS)
+  if(spirv_present_result)
+    set(SPIRV_BACKEND_FOUND TRUE)
+    add_compile_definitions(LLVM_SPIRV_BACKEND_TARGET_PRESENT)
+  endif()
 endif()
 
 set(LLVM_SPIRV_INCLUDE_DIRS ${CMAKE_CURRENT_SOURCE_DIR}/include)
